doc/src/bond_cpp.md

```diff
@@ -185,6 +185,59 @@ particular protocol reader:
 
 See example: `examples/cpp/core/marshaling`.
 
+Type evolution
```

Comment:
It won't let me add a comment where this should be, but at the top, consider linking the phrase "schema versioning rules" to here.